src/java/org/apache/cassandra/index/sai/utils/OrderingFilterRangeIterator.java

```diff
@@ -79,6 +79,9 @@ public PrimaryKey computeNext()
     @Override
     protected void performSkipTo(PrimaryKey nextToken)
     {
+        if (nextIterator == null)
+            tryToComputeNext();
+
         input.skipTo(nextToken);
         nextIterator.skipTo(nextToken);
```

Comment:
#874 adds a null check before calling `nextIterator`. I think that is the correct behavior here. The `input` is a stream of PKs. The `nextIterator` is a stream of the "top" PKs from the stream. Because there is a chunk size on the nextIterator, the "top" PKs are local maxima, and skipping through the result set could produce an incorrect final set of top k.

Reply:
I think I just contradicted myself a bit. I am now questioning if it is valid to call `skipTo` on `nextIterator`. What are the scenarios when we call `skipTo`? I'll need to look closer at the usage to understand the right fix here.